pkgs/test_core/lib/src/util/io.dart

```diff
@@ -43,7 +43,7 @@ final String sdkDir = p.dirname(p.dirname(Platform.resolvedExecutable));
 final OperatingSystem currentOS = (() {
   var name = Platform.operatingSystem;
   var os = OperatingSystem.findByIoName(name);
-  if (os == OperatingSystem.none) return os;
+  if (os != OperatingSystem.none) return os;
```

Comment:
Is this right?

Reply:
It doesn't look right to me.

Why did this need to change? Was there a static error here?

Reply:
If the os is NOT the `none` os, return it.

I think this was a bug in the original migration.

Reply:
hmm, it does look like there was a bug in the migration, but there is _also_ a bug in the non-migrated code.

From what I can tell, `OperatingSystem.findByIoName` never returned null.

We might need to dig a bit deeper on this one.

Reply:
https://github.com/dart-lang/test/pull/1221

Reply:
resolved